src/js/stores/__tests__/MesosStateStore-test.js

```diff
@@ -33,10 +33,102 @@ describe("MesosStateStore", function() {
       expect(result).toEqual([1, 2, 3]);
     });
 
-    it("nulls if no service matches", function() {
+    it("returns empty array if no service matches", function() {
       var result = MesosStateStore.getTasksFromServiceName("nonExistent");
       expect(result).toEqual([]);
     });
+
+    it("returns empty array for an invalid state", function() {
+      MesosStateStore.get = () => null;
+      var result = MesosStateStore.getTasksFromServiceName("marathon");
+      expect(result).toEqual([]);
+    });
+  });
+
+  describe("#getLastMesosState", function() {
+    beforeEach(function() {
+      this.get = MesosStateStore.get;
+    });
+
+    afterEach(function() {
+      MesosStateStore.get = this.get;
+    });
+
+    describe("fills empty fields with empty arrays", function() {
+      const testCases = [
+        [],
+        ["tasks"],
+        ["frameworks", "slaves"],
+        ["frameworks", "tasks", "executors"]
+      ];
+      const allFields = ["tasks", "frameworks", "slaves", "executors"];
+      console.log("ABC");
+
+      testCases.map(function(testCase) {
+        it(`when [${testCase.join(", ")}] are set`, function() {
```

Comment:
Not sure how I feel about using JS dynamic nature to write tests like these.

I'm curious to know what @dcos/frontend peeps think about this?

Reply:
I was thinking the same thing when I saw this. But then I saw we do [elsewhere](https://github.com/dcos/dcos-ui/blob/master/plugins/services/src/js/utils/__tests__/HealthCheckUtil-test.js#L15) so didn't say anything.

Reply:
True. I did a quick iterwebs research and looks like it's a pattern people use for dynamic generating test cases. (I still think it's hacky)

Reply:
In golang it's called table driven tests and the most common test pattern. I personally like it for cases like this, because you don't have to read through that much duplicate code.

If you don't like it I can remove it, I don't feel personally attached to it 🤷‍♂️

Reply:
I think it’s fine I’m just not personally used to see that in tests but makes sense.

Reply:
I left the comment mostly because I was curious to know about the technical details and advantages besides de obvious one of not having duplicated.

Reply:
Ah, I see. Well from my experience with these kinds of tests there are two sides: